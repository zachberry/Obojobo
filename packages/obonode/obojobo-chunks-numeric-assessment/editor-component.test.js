import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import NumericAssessment from './editor-component'

describe('NumericAssessment Editor Node', () => {
	test('NumericAssessment builds the expected component', () => {
		const component = renderer.create(
			<NumericAssessment
				node={{
					data: {
						get: () => {
							return 'mock-question-type'
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericAssessment builds the expected component with defaults', () => {
		const component = renderer.create(
			<NumericAssessment
				node={{
					data: {
						get: () => {
							return null
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericAssessment component adds choice', () => {
		const editor = {
			insertNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericAssessment
				node={{
					key: 'mockKey',
					nodes: [],
					data: {
						get: () => {
							return {}
						}
					}
				}}
				editor={editor}
			/>
		)
		const tree = component.html()

		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(editor.insertNodeByKey).toHaveBeenCalled()
		expect(tree).toMatchSnapshot()
	})
})
