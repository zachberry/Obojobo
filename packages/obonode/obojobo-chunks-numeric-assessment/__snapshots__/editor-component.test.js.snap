// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericAssessment Editor Node NumericAssessment builds the expected component 1`] = `
<div
  className="component obojobo-draft--chunks--numeric-assessment"
>
  <div
    contentEditable={false}
  >
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center add-answer-btn pad"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Add possible answer
      </button>
    </div>
  </div>
</div>
`;

exports[`NumericAssessment Editor Node NumericAssessment builds the expected component with defaults 1`] = `
<div
  className="component obojobo-draft--chunks--numeric-assessment"
>
  <div
    contentEditable={false}
  >
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center add-answer-btn pad"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Add possible answer
      </button>
    </div>
  </div>
</div>
`;

exports[`NumericAssessment Editor Node NumericAssessment component adds choice 1`] = `"<div class=\\"component obojobo-draft--chunks--numeric-assessment\\"><div contenteditable=\\"false\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center add-answer-btn pad\\"><button class=\\"button\\">Add possible answer</button></div></div></div>"`;
